### PR TITLE
change source for coverage report

### PR DESCRIPTION
**Pull request by pstjohn**
Corrects the `source` field for the coverage report, prints out which lines are missed.

### CODE REVIEW

**Review by jstjohn — APPROVED** [review]

File: ci/scripts/run_pytest.sh [hunk 1]
```diff
@@ -29,8 +29,8 @@ python -m coverage erase
 
 for dir in docs/ ./sub-packages/bionemo-*/; do
     echo "Running pytest in $dir"
-    python -m coverage run --parallel-mode --source sub-packages/ -m pytest -v --nbval-lax $dir
+    python -m coverage run --parallel-mode --source bionemo -m pytest -v --nbval-lax $dir
 done
 
 python -m coverage combine
-python -m coverage report
+python -m coverage report -m
```
Comment (trvachov) [inline ci/scripts/run_pytest.sh:36]: Is this really the syntax? `-m ... -m`?

**Review by trvachov — COMMENTED** [review]
Peter could you just copy paste what a sample output looks like from this? Probably it works I just want to double check...

**Review by skothenhill-nv — APPROVED** [review]

File: ci/scripts/run_pytest.sh (hunk 1 above)
Comment (pstjohn) [reply ci/scripts/run_pytest.sh:36]: you're right -- better add a few more -m's for good measure 🤣  

But in all seriousness, yes. Might be less confusing as 
```
python -m coverage report --show-missing
```
but the first invokes coverage (for some reason the `coverage` cli isn't in our path) and the second asks the report to show missing line numbers

**Comment by pstjohn** [comment]
@trvachov Sure thing, this is from https://prod.blsm.nvidia.com/bionemo-external-bionemo-fw/job/test_pytest/966/pipeline-console/?selected-node=66

```
[2024-12-03T21:50:28.653Z] Name                                                                                               Stmts   Miss  Cover   Missing
[2024-12-03T21:50:28.653Z] --------------------------------------------------------------------------------------------------------------------------------
[2024-12-03T21:50:28.653Z] /usr/local/lib/python3.10/dist-packages/bionemo/core/__init__.py                                      15      3    80%   29, 35-36
[2024-12-03T21:50:28.653Z] /usr/local/lib/python3.10/dist-packages/bionemo/core/data/__init__.py                                  0      0   100%
[2024-12-03T21:50:28.653Z] /usr/local/lib/python3.10/dist-packages/bionemo/core/data/load.py                                    126     40    68%   207, 215-222, 227-265, 269, 276-307
[2024-12-03T21:50:28.653Z] /usr/local/lib/python3.10/dist-packages/bionemo/core/data/multi_epoch_dataset.py                      80      6    92%   56, 59, 72, 75, 124, 166
[2024-12-03T21:50:28.653Z] /usr/local/lib/python3.10/dist-packages/bionemo/core/data/permute.py                                  37      0   100%
[2024-12-03T21:50:28.653Z] /usr/local/lib/python3.10/dist-packages/bionemo/core/data/resamplers.py                               33      1    97%   117
[2024-12-03T21:50:28.653Z] /usr/local/lib/python3.10/dist-packages/bionemo/core/data/resource.py                                 55      0   100%
[2024-12-03T21:50:28.653Z] /usr/local/lib/python3.10/dist-packages/bionemo/core/model/__init__.py                                 0      0   100%
[2024-12-03T21:50:28.653Z] /usr/local/lib/python3.10/dist-packages/bionemo/core/model/config.py                                  21      3    86%   46, 60, 69
[2024-12-03T21:50:28.653Z] /usr/local/lib/python3.10/dist-packages/bionemo/core/utils/__init__.py                                 0      0   100%
[2024-12-03T21:50:28.653Z] /usr/local/lib/python3.10/dist-packages/bionemo/core/utils/batching_utils.py                          20      3    85%   51, 68-69
[2024-12-03T21:50:28.653Z] /usr/local/lib/python3.10/dist-packages/bionemo/core/utils/dtypes.py                                  26      3    88%   66, 68, 74
[2024-12-03T21:50:28.653Z] /usr/local/lib/python3.10/dist-packages/bionemo/core/utils/random_utils.py                            13      0   100%
[2024-12-03T21:50:28.653Z] /usr/local/lib/python3.10/dist-packages/bionemo/esm2/__init__.py                                       0      0   100%
[2024-12-03T21:50:28.653Z] /usr/local/lib/python3.10/dist-packages/bionemo/esm2/api.py                                            3      0   100%
[2024-12-03T21:50:28.653Z] /usr/local/lib/python3.10/dist-packages/bionemo/esm2/data/__init__.py                                  0      0   100%
[2024-12-03T21:50:28.653Z] /usr/local/lib/python3.10/dist-packages/bionemo/esm2/data/datamodule.py                               60      3    95%   114, 131, 218
[2024-12-03T21:50:28.653Z] /usr/local/lib/python3.10/dist-packages/bionemo/esm2/data/dataset.py                                  96     10    90%   86, 151, 258, 261, 265, 294, 298, 338-339, 342
[2024-12-03T21:50:28.653Z] /usr/local/lib/python3.10/dist-packages/bionemo/esm2/data/tokenizer/__init__.py                       11      0   100%
[2024-12-03T21:50:28.653Z] /usr/local/lib/python3.10/dist-packages/bionemo/esm2/model/__init__.py                                 0      0   100%
[2024-12-03T21:50:28.654Z] /usr/local/lib/python3.10/dist-packages/bionemo/esm2/model/attention.py                               97     50    48%   64, 74-75, 102, 107-108, 116-119, 129, 181, 212-319, 342-365
[2024-12-03T21:50:28.654Z] /usr/local/lib/python3.10/dist-packages/bionemo/esm2/model/embedding.py                               56     12    79%   118-119, 131-135, 141, 145-152
[2024-12-03T21:50:28.654Z] /usr/local/lib/python3.10/dist-packages/bionemo/esm2/model/finetune/__init__.py                        0      0   100%
[2024-12-03T21:50:28.654Z] /usr/local/lib/python3.10/dist-packages/bionemo/esm2/model/finetune/datamodule.py                     98      4    96%   112, 216, 225, 300
[2024-12-03T21:50:28.654Z] /usr/local/lib/python3.10/dist-packages/bionemo/esm2/model/finetune/finetune_regressor.py             85      9    89%   75, 144, 147, 214-219, 237-238
[2024-12-03T21:50:28.654Z] /usr/local/lib/python3.10/dist-packages/bionemo/esm2/model/finetune/finetune_token_classifier.py     104     15    86%   96, 166, 169, 240-246, 256-268, 279-280
[2024-12-03T21:50:28.654Z] /usr/local/lib/python3.10/dist-packages/bionemo/esm2/model/finetune/peft.py                           13      6    54%   49-51, 67-69
[2024-12-03T21:50:28.654Z] /usr/local/lib/python3.10/dist-packages/bionemo/esm2/model/finetune/train.py                          52     14    73%   113, 120, 152-189
[2024-12-03T21:50:28.654Z] /usr/local/lib/python3.10/dist-packages/bionemo/esm2/model/model.py                                  127      9    93%   111, 196-200, 238, 350-357
[2024-12-03T21:50:28.654Z] /usr/local/lib/python3.10/dist-packages/bionemo/esm2/scripts/__init__.py                               0      0   100%
[2024-12-03T21:50:28.654Z] /usr/local/lib/python3.10/dist-packages/bionemo/esm2/scripts/infer_esm2.py                            65     27    58%   83, 148-151, 171-270, 274
[2024-12-03T21:50:28.654Z] /usr/local/lib/python3.10/dist-packages/bionemo/esm2/scripts/train_esm2.py                            99      7    93%   196-198, 302-305, 658
[2024-12-03T21:50:28.654Z] /usr/local/lib/python3.10/dist-packages/bionemo/example_model/lightning/__init__.py                    0      0   100%
[2024-12-03T21:50:28.654Z] /usr/local/lib/python3.10/dist-packages/bionemo/example_model/lightning/lightning_basic.py           205      3    99%   325, 613, 637
[2024-12-03T21:50:28.654Z] /usr/local/lib/python3.10/dist-packages/bionemo/geneformer/__init__.py                                 0      0   100%
[2024-12-03T21:50:28.654Z] /usr/local/lib/python3.10/dist-packages/bionemo/geneformer/api.py                                     46      0   100%
[2024-12-03T21:50:28.654Z] /usr/local/lib/python3.10/dist-packages/bionemo/geneformer/data/__init__.py                            0      0   100%
[2024-12-03T21:50:28.654Z] /usr/local/lib/python3.10/dist-packages/bionemo/geneformer/data/preprocess.py                         19      4    79%   39, 42, 47, 52
[2024-12-03T21:50:28.654Z] /usr/local/lib/python3.10/dist-packages/bionemo/geneformer/data/singlecell/__init__.py                 0      0   100%
[2024-12-03T21:50:28.654Z] /usr/local/lib/python3.10/dist-packages/bionemo/geneformer/data/singlecell/datamodule.py              83     14    83%   91-95, 148-161, 166-167, 190, 229-230, 246
[2024-12-03T21:50:28.654Z] /usr/local/lib/python3.10/dist-packages/bionemo/geneformer/data/singlecell/dataset.py                136     39    71%   144-145, 151-169, 183-185, 194, 288, 291, 296, 351-372, 377
[2024-12-03T21:50:28.654Z] /usr/local/lib/python3.10/dist-packages/bionemo/geneformer/data/singlecell/preprocess.py              54      2    96%   119, 130
[2024-12-03T21:50:28.654Z] /usr/local/lib/python3.10/dist-packages/bionemo/geneformer/data/singlecell/utils.py                    8      2    75%   38-39
[2024-12-03T21:50:28.654Z] /usr/local/lib/python3.10/dist-packages/bionemo/geneformer/model/__init__.py                           0      0   100%
[2024-12-03T21:50:28.654Z] /usr/local/lib/python3.10/dist-packages/bionemo/geneformer/model/finetune_token_regressor.py         128     50    61%   93, 113, 120-141, 190, 193, 244-250, 254-260, 264-266, 270-272, 278-321
[2024-12-03T21:50:28.654Z] /usr/local/lib/python3.10/dist-packages/bionemo/geneformer/scripts/__init__.py                         0      0   100%
[2024-12-03T21:50:28.654Z] /usr/local/lib/python3.10/dist-packages/bionemo/geneformer/scripts/train_geneformer.py               121     15    88%   174, 221, 226-228, 256-257, 339-342, 554-555, 629-633, 678
[2024-12-03T21:50:28.654Z] /usr/local/lib/python3.10/dist-packages/bionemo/geneformer/tokenizer/__init__.py                       0      0   100%
[2024-12-03T21:50:28.654Z] /usr/local/lib/python3.10/dist-packages/bionemo/geneformer/tokenizer/gene_tokenizer.py                89     25    72%   65, 100, 106, 118-127, 138, 149, 163-169, 183-189
[2024-12-03T21:50:28.654Z] /usr/local/lib/python3.10/dist-packages/bionemo/geometric/__init__.py                                  0      0   100%
[2024-12-03T21:50:28.654Z] /usr/local/lib/python3.10/dist-packages/bionemo/geometric/atom_featurizers.py                        179     18    90%   51, 73, 95, 121, 147, 173, 195, 223, 233, 282, 287, 292, 355, 360, 365, 392, 432, 485
[2024-12-03T21:50:28.654Z] /usr/local/lib/python3.10/dist-packages/bionemo/geometric/base_featurizer.py                          43     15    65%   30, 35, 48, 53, 66, 71, 94-96, 114-116, 134-139
[2024-12-03T21:50:28.654Z] /usr/local/lib/python3.10/dist-packages/bionemo/geometric/bond_featurizers.py                         14      1    93%   37
[2024-12-03T21:50:28.654Z] /usr/local/lib/python3.10/dist-packages/bionemo/geometric/molecule_featurizers.py                     11      1    91%   40
[2024-12-03T21:50:28.654Z] /usr/local/lib/python3.10/dist-packages/bionemo/llm/__init__.py                                        0      0   100%
[2024-12-03T21:50:28.654Z] /usr/local/lib/python3.10/dist-packages/bionemo/llm/api.py                                            11      1    91%   41
[2024-12-03T21:50:28.654Z] /usr/local/lib/python3.10/dist-packages/bionemo/llm/data/__init__.py                                   0      0   100%
[2024-12-03T21:50:28.654Z] /usr/local/lib/python3.10/dist-packages/bionemo/llm/data/collate.py                                   31      0   100%
[2024-12-03T21:50:28.654Z] /usr/local/lib/python3.10/dist-packages/bionemo/llm/data/datamodule.py                                24      3    88%   58-60
[2024-12-03T21:50:28.654Z] /usr/local/lib/python3.10/dist-packages/bionemo/llm/data/label2id_tokenizer.py                        46     20    57%   51, 54, 64-71, 81-87, 98-99, 102-103
[2024-12-03T21:50:28.654Z] /usr/local/lib/python3.10/dist-packages/bionemo/llm/data/masking.py                                   38      2    95%   71, 74
[2024-12-03T21:50:28.654Z] /usr/local/lib/python3.10/dist-packages/bionemo/llm/data/types.py                                     17      2    88%   53, 57
[2024-12-03T21:50:28.654Z] /usr/local/lib/python3.10/dist-packages/bionemo/llm/lightning.py                                     155     11    93%   81, 278, 283-286, 326, 411, 433, 437-438
[2024-12-03T21:50:28.654Z] /usr/local/lib/python3.10/dist-packages/bionemo/llm/model/__init__.py                                  0      0   100%
[2024-12-03T21:50:28.654Z] /usr/local/lib/python3.10/dist-packages/bionemo/llm/model/biobert/__init__.py                          0      0   100%
[2024-12-03T21:50:28.654Z] /usr/local/lib/python3.10/dist-packages/bionemo/llm/model/biobert/lightning.py                        81     25    69%   63, 117, 143, 195, 214, 217-237, 259-271, 298
[2024-12-03T21:50:28.654Z] /usr/local/lib/python3.10/dist-packages/bionemo/llm/model/biobert/model.py                           226     19    92%   184, 225, 253-254, 281-284, 355, 358, 387-388, 401, 427, 442, 454, 458, 537-538, 546, 592
[2024-12-03T21:50:28.654Z] /usr/local/lib/python3.10/dist-packages/bionemo/llm/model/biobert/transformer_specs.py                52      5    90%   157-191, 195
[2024-12-03T21:50:28.654Z] /usr/local/lib/python3.10/dist-packages/bionemo/llm/model/config.py                                   42      2    95%   122, 126
[2024-12-03T21:50:28.654Z] /usr/local/lib/python3.10/dist-packages/bionemo/llm/model/layers.py                                   19      1    95%   37
[2024-12-03T21:50:28.654Z] /usr/local/lib/python3.10/dist-packages/bionemo/llm/model/loss.py                                     91     34    63%   76-98, 108-109, 124-137, 183, 214, 221-246, 279, 290
[2024-12-03T21:50:28.654Z] /usr/local/lib/python3.10/dist-packages/bionemo/llm/model/lr_scheduler.py                             43      0   100%
[2024-12-03T21:50:28.654Z] /usr/local/lib/python3.10/dist-packages/bionemo/llm/utils/__init__.py                                  0      0   100%
[2024-12-03T21:50:28.654Z] /usr/local/lib/python3.10/dist-packages/bionemo/llm/utils/datamodule_utils.py                         46      8    83%   89, 94, 96, 98, 100, 102, 104, 108
[2024-12-03T21:50:28.654Z] /usr/local/lib/python3.10/dist-packages/bionemo/llm/utils/iomixin_utils.py                            26      3    88%   50, 63, 72
[2024-12-03T21:50:28.654Z] /usr/local/lib/python3.10/dist-packages/bionemo/llm/utils/logger_utils.py                             30      2    93%   82, 87
[2024-12-03T21:50:28.654Z] /usr/local/lib/python3.10/dist-packages/bionemo/llm/utils/megatron_utils.py                            8      0   100%
[2024-12-03T21:50:28.654Z] /usr/local/lib/python3.10/dist-packages/bionemo/llm/utils/remote.py                                   57     12    79%   118-122, 139, 142, 151-157
[2024-12-03T21:50:28.654Z] /usr/local/lib/python3.10/dist-packages/bionemo/llm/utils/weight_utils.py                             53      2    96%   55, 79
[2024-12-03T21:50:28.654Z] /usr/local/lib/python3.10/dist-packages/bionemo/noodles/__init__.py                                    2      0   100%
[2024-12-03T21:50:28.654Z] /usr/local/lib/python3.10/dist-packages/bionemo/noodles/nvfaidx.py                                    63     16    75%   65, 74, 78, 85, 120, 122, 125, 127, 131-135, 148, 151, 163-165
[2024-12-03T21:50:28.654Z] /usr/local/lib/python3.10/dist-packages/bionemo/scdl/__init__.py                                       0      0   100%
[2024-12-03T21:50:28.654Z] /usr/local/lib/python3.10/dist-packages/bionemo/scdl/api/__init__.py                                   0      0   100%
[2024-12-03T21:50:28.654Z] /usr/local/lib/python3.10/dist-packages/bionemo/scdl/api/single_cell_row_dataset.py                   39      9    77%   39, 44, 49, 54, 63, 74, 87, 100, 105
[2024-12-03T21:50:28.654Z] /usr/local/lib/python3.10/dist-packages/bionemo/scdl/index/__init__.py                                 0      0   100%
[2024-12-03T21:50:28.654Z] /usr/local/lib/python3.10/dist-packages/bionemo/scdl/index/row_feature_index.py                       97      4    96%   146, 223-224, 227
[2024-12-03T21:50:28.654Z] /usr/local/lib/python3.10/dist-packages/bionemo/scdl/io/__init__.py                                    0      0   100%
[2024-12-03T21:50:28.654Z] /usr/local/lib/python3.10/dist-packages/bionemo/scdl/io/single_cell_collection.py                     82      2    98%   157, 183
[2024-12-03T21:50:28.654Z] /usr/local/lib/python3.10/dist-packages/bionemo/scdl/io/single_cell_memmap_dataset.py                286     20    93%   88, 188, 191, 281, 284, 415, 421, 433, 444, 455-456, 499, 505, 547, 651, 658, 682, 747-748, 754
[2024-12-03T21:50:28.654Z] /usr/local/lib/python3.10/dist-packages/bionemo/scdl/util/__init__.py                                  0      0   100%
[2024-12-03T21:50:28.654Z] /usr/local/lib/python3.10/dist-packages/bionemo/scdl/util/async_worker_queue.py                       45      0   100%
[2024-12-03T21:50:28.654Z] /usr/local/lib/python3.10/dist-packages/bionemo/scdl/util/torch_dataloader_utils.py                   10      0   100%
[2024-12-03T21:50:28.654Z] /usr/local/lib/python3.10/dist-packages/bionemo/size_aware_batching/__init__.py                        0      0   100%
[2024-12-03T21:50:28.654Z] /usr/local/lib/python3.10/dist-packages/bionemo/size_aware_batching/sampler.py                       156      6    96%   165, 436, 591-594
[2024-12-03T21:50:28.654Z] /usr/local/lib/python3.10/dist-packages/bionemo/size_aware_batching/utils.py                          74      1    99%   208
[2024-12-03T21:50:28.654Z] /usr/local/lib/python3.10/dist-packages/bionemo/testing/__init__.py                                    0      0   100%
[2024-12-03T21:50:28.654Z] /usr/local/lib/python3.10/dist-packages/bionemo/testing/callbacks.py                                  36      6    83%   35-36, 40, 55-57
[2024-12-03T21:50:28.654Z] /usr/local/lib/python3.10/dist-packages/bionemo/testing/data/__init__.py                               0      0   100%
[2024-12-03T21:50:28.654Z] /usr/local/lib/python3.10/dist-packages/bionemo/testing/data/esm2.py                                  20      0   100%
[2024-12-03T21:50:28.654Z] /usr/local/lib/python3.10/dist-packages/bionemo/testing/harnesses/__init__.py                          0      0   100%
[2024-12-03T21:50:28.654Z] /usr/local/lib/python3.10/dist-packages/bionemo/testing/harnesses/mode.py                              5      0   100%
[2024-12-03T21:50:28.654Z] /usr/local/lib/python3.10/dist-packages/bionemo/testing/harnesses/stop_and_go.py                     116      1    99%   157
[2024-12-03T21:50:28.654Z] /usr/local/lib/python3.10/dist-packages/bionemo/testing/lightning.py                                  10      0   100%
[2024-12-03T21:50:28.654Z] /usr/local/lib/python3.10/dist-packages/bionemo/testing/megatron_dataset_compatibility.py             35      1    97%   41
[2024-12-03T21:50:28.654Z] /usr/local/lib/python3.10/dist-packages/bionemo/testing/megatron_parallel_state_utils.py             128     10    92%   163-164, 267, 277-278, 286, 340-342, 348
[2024-12-03T21:50:28.654Z] /usr/local/lib/python3.10/dist-packages/bionemo/testing/testing_callbacks.py                          88      2    98%   227, 234
[2024-12-03T21:50:28.654Z] /usr/local/lib/python3.10/dist-packages/bionemo/testing/torch.py                                      24      0   100%
[2024-12-03T21:50:28.654Z] /usr/local/lib/python3.10/dist-packages/bionemo/testing/utils.py                                      23      4    83%   48, 62, 85, 93
[2024-12-03T21:50:28.654Z] /usr/local/lib/python3.10/dist-packages/bionemo/webdatamodule/__init__.py                              0      0   100%
[2024-12-03T21:50:28.654Z] /usr/local/lib/python3.10/dist-packages/bionemo/webdatamodule/datamodule.py                           86      5    94%   200, 209, 243, 275, 290
[2024-12-03T21:50:28.654Z] /usr/local/lib/python3.10/dist-packages/bionemo/webdatamodule/utils.py                                27      6    78%   81, 87, 113-119
[2024-12-03T21:50:28.654Z] --------------------------------------------------------------------------------------------------------------------------------
[2024-12-03T21:50:28.654Z] TOTAL                                                                                               5076    659    87%
```

File: ci/scripts/run_pytest.sh (hunk 1 above)
Comment (malcolmgreaves) [reply ci/scripts/run_pytest.sh:36]: +1 to use `--show-missing` instead of `-m`

**Review by malcolmgreaves — APPROVED** [review]
LGTM. Suggest making the coverage report command clearer.

**Comment by trvachov** [comment]
Late to approve, but thanks for merging (my comments were NITs mostly).

**Pull request merged**